lib/CFRPageActions.jsm

```diff
@@ -0,0 +1,241 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const LABEL_TEXT = "Firefox recommends";
+const BUTTON_ICON = "resource://activity-stream/data/content/assets/glyph-webextension-16.svg";
+const POPUP_NOTIFICATION_ID = "contextual-feature-recommendation";
+
+const ACTION_TYPE_INSTALL_ADDON = "ACTION_TYPE_INSTALL_ADDON";
+
+/**
+ * A WeakMap from browsers to recommendations. A recommendation is an object
+ * with the following properties:
+ * - host:            the hostname to target
+ * - message:         the message to be shown
+ * - icon:            a uri to the icon to be shown
+ * - name (optional): a name to replace (in bold) the substring "<>" in the message
+ * - action:
+ *     - type:          the type of action being recommended
+ *     - data:          data specific to the action type
+ *
+ * A recommendation is specific to a browser and host and is active until the
+ * given browser is closed or the user navigates (within that browser) away from
+ * the host.
+ */
+const RecommendationMap = new WeakMap();
+
+/**
+ * A WeakMap from windows to their CFR PageAction.
+ */
+const PageActionMap = new WeakMap();
+
+/**
+ * We need one PageAction for each window
+ */
+class PageAction {
+  constructor(win) {
+    this.window = win;
+    this.hbox = win.document.getElementById("contextual-feature-recommendation");
```

Comment:
Could you give this a name that reflects its specific function rather than just `hbox`?